tests/tests/loopback.rs

```diff
@@ -0,0 +1,306 @@
+use anyhow::Result;
+use opteadm::OpteAdm;
+use oxide_vpc::api::{
+    AddRouterEntryReq, Address, BoundaryServices, Direction, Filters,
+    FirewallAction, FirewallRule, IpCfg, IpCidr, Ipv4Cfg, PhysNet, Ports,
+    RouterTarget, SNat4Cfg, SetVirt2PhysReq, Vni, VpcCfg,
+};
+use std::env;
+use std::process::Command;
+use ztest::*;
+
+/// This is a wrapper around the ztest::Zone object that encapsulates common
+/// logic needed for running the OPTE tests zones used in this test suite.
+struct OpteZone<'a> {
+    _zfs: &'a Zfs,
+    zone: Zone,
+}
+
+impl<'a> OpteZone<'a> {
+    /// Create a new zone with the given name, underlying zfs instance and set
+    /// of interfaces. In illumos parlance, the interfaces are data link
+    /// devices.
+    fn new(name: &str, zfs: &'a Zfs, ifx: &[&'a str]) -> Result<Self> {
+        // If we're running in CI, use the omicron1 zone, because that's whats
+        // there. If this is running locally, use a sparse zone which is much
+        // easier to set up on a regular Helios dev box.
+        let brand = match env::var("BUILDOMAT_JOB_ID") {
+            Ok(_) => "omicron1",
+            _ => "sparse",
+        };
+        let zone = Zone::new(name, brand, zfs, ifx)?;
+        Ok(Self { _zfs: zfs, zone })
+    }
+
+    /// Wait for the network to come up, then set up the overlay network.
+    fn setup(&self, devname: &str, addr: String) -> Result<()> {
+        self.zone.wait_for_network()?;
+        self.zone
+            .zexec(&format!("ipadm create-addr -t -T dhcp {}/test", devname))?;
+        self.zone.zexec(&format!("route add -iface 10.0.0.254 {}", addr))?;
+        self.zone.zexec("route add 10.0.0.0/24 10.0.0.254")?;
```

Comment:
Where do these come from? Should we pull them to `const`s?

Reply:
I believe it's based on the `OptePort`'s `VpcCfg` below. The fact we have to add these routes here manually is also just a bug in the illumos DHCP client (https://github.com/oxidecomputer/stlouis/issues/326 / https://www.illumos.org/issues/11990).

Reply:
They are the overlay network subnet/gw used in tests. Added constants and comments.